esp/esp/utils/widgets.py

```diff
@@ -330,15 +331,311 @@ class NavStructureWidget(forms.Widget):
 </style>
 """
 
+    # We separate out this part so subclasses can override it.
+    add_link_body = """
+        entry.append($j("<span>Text: </span>"));
+        entry.append($j("<input class='data_text nav_secondary_field input-small' type='text' value='" + data.text + "' />"));
+        entry.append($j("<span>Link: </span>"));
+        entry.append($j("<input class='data_link nav_secondary_field' type='text' value='" + data.link + "' />"));
+    """
+
     def render(self, name, value, attrs=None):
         if value is None: value = ''
         final_attrs = self.build_attrs(attrs, name=name)
         context = {}
         context['name'] = name
         context['value'] = json.dumps(value)
-        template = Template(NavStructureWidget.template_text)
+        template = Template(self.template_text % {
+            'add_link_body': self.add_link_body})
         return template.render(Context(context))
 
     def value_from_datadict(self, data, files, name):
         result = json.loads(data[name])
         return result
+
+
+_ICONS = [  # copied from http://getbootstrap.com/components/
```

Comment:
Oh hmm.  Actually it probably makes sense to just make that switch now so we don't have to worry about breaking things.  Will do.